specification/dsdl/versioning.tex

```diff
@@ -0,0 +1,411 @@
+        uint8 FLAG_PSU_MALFUNCTION = 32
+        uint8 FLAG_OVERHEATING     = 64
+        uint8 FLAG_CRYOBOX_BREACH  = 128
+        # Storage for the above defined flags (this is not the recommended practice)
+        uint8 flags
+    \end{minted}
+
+    The last definition is deemed sufficient and is deployed to the production system
+    under the version number of 1.0: \verb|sirius_cyber_corp.b_ark.cryopod.Status.1.0|.
+
+    Having collected empirical data from the fielded systems, the Sirius Cybernetics Corporation has
+    identified a shortcoming in the v1.0 definition, which is corrected in an updated definition.
+    Since the updated definition, which is shown below, is mutually semantically
+    compatible\footnote{The topic of data serialization is explored in detail in the section
+    \ref{sec:dsdl_data_serialization}.}
+    with v1.0, the major version number is kept the same and the minor version number is incremented by one:
+
+    \begin{minted}{python}
+        # sirius_cyber_corp.b_ark.cryopod.Status.1.1
+
+        saturated float16 internal_temperature    # [kelvin]
+        saturated float16 coolant_temperature     # [kelvin]
+
+        float32[3] power_consumption    # [watt] Power consumption by the PSU
+
+        # Error flags (this is the recommended practice)
+        bool flag_cryobox_breach
+        bool flag_overheating
+        bool flag_psu_malfunction
+
+        void3   # Reserved for other flags
+
+        # Status flags (this is the recommended practice)
+        bool flag_cooling_system_a_active
+        bool flag_cooling_system_b_active
+    \end{minted}
+
+    Since the definitions v1.0 and v1.1 are mutually bit-compatible and semantically compatible,
+    UAVCAN nodes using either of them can successfully interoperate on the same bus.
+
+    Suppose further that at some point a newer version of the cryopod module,
+    equipped with better temperature sensors, is released.
+    The definition is updated accordingly to use \verb|float32| for the temperature fields instead of \verb|float16|.
+    Seeing as that change breaks the bit compatibility,
+    the major version number has to be incremented by one,
+    and the minor version number has to be reset back to zero:
+
+    \begin{minted}{python}
+        # sirius_cyber_corp.b_ark.cryopod.Status.2.0
+
+        float32 internal_temperature    # [kelvin]
+        float32 coolant_temperature     # [kelvin]
+
+        float32[3] power_consumption    # [watt] Power consumption by the PSU
+
+        # Error flags (this is the recommended practice)
+        bool flag_cryobox_breach
+        bool flag_overheating
+        bool flag_psu_malfunction
+
+        void3   # Reserved for other flags
+
+        # Status flags (this is the recommended practice)
+        bool flag_cooling_system_a_active
+        bool flag_cooling_system_b_active
+    \end{minted}
+
+    Nodes using v1.0, v1.1, and v2.0 definitions can still coexist on the same network,
```

Comment:
Please describe how the nodes determine what the major version is of a given message and how they must use this to lookup the correct response type major version.

Reply:
I just added a reference to the relevant section in the Basic Concepts: https://github.com/UAVCAN/specification/pull/46/commits/c9ccffb145b167a92bc06df10bf503d94f0374e3